Cargo.toml

```diff
@@ -30,12 +30,12 @@ base64 = "0.21.3"
 bitvec = { version = "1.0.1", features = ["serde"] }
 # Point to the latest commit of branch msl/backport-secp-patches-0.8.0-rc.3
 blockifier = { git = "https://github.com/Moonsong-Labs/sequencer", rev = "6624e910c57db9a16f1607c1ed26f7d8f1114e73", features = ["testing"] }
-cairo-lang-starknet = { version = "=2.7.1" }
-cairo-lang-starknet-classes = { version = "=2.7.1" }
-cairo-lang-utils = { version = "=2.7.1" }
-cairo-lang-casm = { version = "=2.7.1" }
+cairo-lang-starknet = { version = "=2.8.2" }
+cairo-lang-starknet-classes = { version = "=2.8.2" }
+cairo-lang-utils = { version = "=2.8.2" }
+cairo-lang-casm = { version = "=2.8.2" }
 cairo-type-derive = { version = "0.1.0", path = "crates/cairo-type-derive" }
-cairo-vm = { git = "https://github.com/Moonsong-Labs/cairo-vm", rev = "56b68b50944ecb3123a168218ea7b8b8e23f9be8", features = ["cairo-1-hints", "extensive_hints", "mod_builtin"] }
+cairo-vm = { git = "https://github.com/Moonsong-Labs/cairo-vm", branch = "notlesh/segment-arena-relocation-fix", features = ["cairo-1-hints", "extensive_hints", "mod_builtin"] }
```

Comment:
Did you bump this branch to cairo-vm latest? 
@notlesh

Reply:
Yes, it the change here basically does:

1. undo the previous hack mentioned in the PR description (by not including the commit 56b68b5)
2. include `merge main` which pulls in the upstream fix (also mentioned in the PR desc)